app/src/main/java/org/greenstand/android/TreeTracker/activities/TreeTrackerActivity.kt

```diff
@@ -26,11 +37,36 @@ class TreeTrackerActivity : ComponentActivity() {
         } else {
             languageSwitcher.applyCurrentLanguage(this)
         }
-
-        setContent {
-            CustomTheme {
-                Root(viewModelFactory)
+        if (hasGPSDevice(this)) {
```

Comment:
It looks like the main difference here is Root vs NoGPSDiviceDialog. We can move the if statement inside the custom theme.